### PR TITLE
Add UpdateRole rpc and an expired flag to Invitations

### PR DESCRIPTION
**Pull request by rdimitrov**
# Summary

The following PR:
* adds an UpdateRole RPC method
* adds an expired flag to the Invitations message

Related to: https://github.com/stacklok/epics/issues/212

## Change Type

***Mark the type of change your PR introduces:***

- [ ] Bug fix (resolves an issue without affecting existing features)
- [x] Feature (adds new functionality without breaking changes)
- [ ] Breaking change (may impact existing functionalities or require documentation updates)
- [ ] Documentation (updates or additions to documentation)
- [ ] Refactoring or test improvements (no bug fixes or new functionality)

# Testing

***Outline how the changes were tested, including steps to reproduce and any relevant configurations. 
Attach screenshots if helpful.***

# Review Checklist:

- [x] Reviewed my own code for quality and clarity.
- [ ] Added comments to complex or tricky code sections.
- [ ] Updated any affected documentation.
- [ ] Included tests that validate the fix or feature.
- [ ] Checked that related changes are merged.

### CODE REVIEW

**Review by dmjb — APPROVED** [review]

File: proto/minder/v1/minder.proto [hunk 1]
```diff
@@ -2038,6 +2051,19 @@ message AssignRoleResponse {
     Invitation invitation = 2;
 }
 
+message UpdateRoleRequest {
+    // context is the context in which the role assignment is evaluated.
+    Context context = 1;
+    // role_assignments are the role assignments to be updated.
+    repeated RoleAssignment role_assignments = 2;
+}
```
Comment (evankanderson) [inline proto/minder/v1/minder.proto:2059]: I'm confused by the semantics of this RPC.  If my ListAssignments has:

```json
[
  {"joe", "admin"},
  {"fred", "editor"},
  {"sue", "admin"}
]
```

And I call UpdateRoleRequest with the following payload:

```json
[
  {"joe", "editor"},
  {"fred", "viewer"}
]
```

What's the expected outcome?

I'd expect something like:

```suggestion
message UpdateRoleRequest {
    // context is the context in which the role assignment is evaluated.
    Context context = 1;
    // subject is the account to change permissions for.
    // The account must already have permissions on the project
    subject string = 2;
    // All subject roles are _replaced_ with the following role assignments.  Must be non-empty,
    // use RemoveRole to remove permissions entirely from the project.
    repeated role string = 3;
}
```

Comment (evankanderson) [reply proto/minder/v1/minder.proto:2059]: Repeating subject and project multiple times just seems like an invitation for funny business, so I'd just use direct fields here, unless this is going to be a PATCH method (you probably don't want that given our structures).

**Review by evankanderson — COMMENTED** [review]
As mentioned offline, the `expired` boolean makes sense because the client time and server time may differ.

File: proto/minder/v1/minder.proto [hunk 2]
```diff
@@ -67,6 +67,7 @@ enum Relation {
     RELATION_PROFILE_STATUS_GET = 33 [(name) = "profile_status_get"];
     RELATION_REMOTE_REPO_GET = 34 [(name) = "remote_repo_get"];
     RELATION_ENTITY_RECONCILIATION_TASK_CREATE = 35 [(name) = "entity_reconciliation_task_create"];
+    RELATION_ROLE_ASSIGNMENT_UPDATE = 36 [(name) = "role_assignment_update"];
```
[the remark this answers is not in the document]
Comment (rdimitrov) [reply proto/minder/v1/minder.proto:70]: Ah, thanks!

File: proto/minder/v1/minder.proto (hunk 1 above)
Comment (rdimitrov) [reply proto/minder/v1/minder.proto]: oh, that's better, thanks! 👍 

You've switched the places of type and field name in the suggestion so I applied it separately

**Review by eleftherias — APPROVED** [review]

**Pull request merged**